frame/support/procedural/src/pallet/parse/storage.rs

```diff
@@ -76,12 +79,45 @@ impl syn::parse::Parse for PalletStorageAttr {
 			})?;
 
 			Ok(Self::StorageName(renamed_prefix, attr_span))
+		} else if lookahead.peek(keyword::unbounded) {
+			content.parse::<keyword::unbounded>()?;
+
+			Ok(Self::Unbounded(attr_span))
 		} else {
 			Err(lookahead.error())
 		}
 	}
 }
 
+struct PalletStorageAttrInfo {
+	getter: Option<syn::Ident>,
+	rename_as: Option<syn::LitStr>,
+	unbounded: bool,
+}
+
+impl PalletStorageAttrInfo {
+	fn from_attrs(attrs: Vec<PalletStorageAttr>) -> syn::Result<Self> {
+		let mut getter = None;
+		let mut rename_as = None;
+		let mut unbounded = false;
+		for attr in attrs {
+			match attr {
+				PalletStorageAttr::Getter(ident, ..) if getter.is_none() => getter = Some(ident),
+				PalletStorageAttr::StorageName(name, ..) if rename_as.is_none() =>
```

Comment:
This seems like a change in api: before we errored if there was more than 1 `getter` or `name`. Now we simply skip anything after the first one.

for example we had

```rust
		if names.len() > 1 {
			let msg = "Invalid pallet::storage, multiple argument pallet::storage_prefix found";
			return Err(syn::Error::new(names[1].attr_span(), msg))
		}
```

Reply:
looking at the UI tests though it seems like it still complains about a duplicate attribute, is this just now an error we can rely on the compiler for?

Reply:
the duplicate attribute is not from the compiler it is from the last matching arm below, in case rename_as is already some, then the last matching are is branched and the error is raised